src/components/ReactTags.js

```diff
@@ -105,6 +105,15 @@ class ReactTags extends Component {
     this.handleSuggestionClick = this.handleSuggestionClick.bind(this);
   }
 
+  static getDerivedStateFromProps(props)
+  {
+    const { classNames } = props;
+
+    return {
+      classNames : {...DEFAULT_CLASSNAMES,...classNames},
```

Comment:
move `...DEFAULT_CLASSNAMES`  to constructor.
use [memoize-one](https://www.google.com/search?q=memoize_one&oq=memo&aqs=chrome.0.69i59j69i57j69i60j0l3.1328j1j7&sourceid=chrome&ie=UTF-8) here so that state changes only when className prop changes.